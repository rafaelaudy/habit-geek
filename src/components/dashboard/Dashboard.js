import React, { useState } from "react";

import SaveHabit from "../habits/SaveHabit";
import DashboardView from "./DashboardView";

const Dashboard = ({
  username,
  currentHabits,
  previousHabits,
  currentWeek,
  previousWeek,
  saveHabit,
  deleteHabit,
  toggleDayHabit
}) => {
  const [isEditingHabit, setIsEditingHabit] = useState({});

  if (isEditingHabit.isEditing) {
    const { name, frequency, type } = isEditingHabit.id
      ? currentHabits.filter(({ name }) => name === isEditingHabit.id)[0]
      : {};

    return (
      <SaveHabit
        id={isEditingHabit.id}
        name={name}
        frequency={frequency}
        type={type}
        saveHabit={saveHabit}
        deleteHabit={deleteHabit}
        onGoBack={() => setIsEditingHabit({})}
      />
    );
  }

  return (
    <DashboardView
      username={username}
      currentHabits={currentHabits}
      previousHabits={previousHabits}
      currentWeek={currentWeek}
      previousWeek={previousWeek}
      onAddNewHabit={() => setIsEditingHabit({ isEditing: true })}
      onUpdateHabit={id => setIsEditingHabit({ isEditing: true, id })}
      toggleDayHabit={toggleDayHabit}
    />
  );
};

export default Dashboard;
