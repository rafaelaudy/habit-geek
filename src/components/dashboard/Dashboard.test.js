import React from "react";
import { shallow, mount } from "enzyme";
import Dashboard from "./Dashboard";

const props = {
  username: "rafa",
  currentHabits: [],
  previousHabits: [],
  currentWeek: "y1w1",
  previousWeek: "y1w0",
  saveHabit: () => {},
  deleteHabit: () => {},
  toggleDayHabit: () => {}
};

jest.mock("@rafael.audy/habit-geek-utils/utils/dateUtils.js", () => ({
  getTodayIndex: jest.fn().mockReturnValue(1),
  getCurrentWeek: jest.fn().mockReturnValue("y1w1"),
  getWeekIntervalText: jest.fn().mockReturnValue("30/Sep ... 6/Oct"),
  isToday: jest.fn().mockReturnValue(false)
}));

describe("Dashboard component", () => {
  it("Renders Dashboard list", () => {
    const dashboard = shallow(<Dashboard {...props} />);
    expect(dashboard).toMatchSnapshot();
  });

  it("Renders New Habit", () => {
    const dashboard = mount(<Dashboard {...props} />);
    dashboard.find(".btn-primary").simulate("click");
    expect(dashboard).toMatchSnapshot();
  });

  it("Updates habit", () => {
    const dashboard = mount(
      <Dashboard
        currentWeek="y1w1"
        currentHabits={[
          { name: "write", frequency: "1", type: "health", checked: [] }
        ]}
      />
    );
    dashboard.find(".habit-row .btn").simulate("click");
    expect(dashboard.find("#new-habit-name").props().value).toBe("write");
    expect(dashboard.find("#new-habit-type").props().value).toBe("health");
    expect(dashboard.find("#new-habit-frequency").props().value).toBe("1");
  });

  it("Comes back from new/update habit", () => {
    const dashboard = mount(
      <Dashboard
        currentHabits={[
          { name: "write", frequency: "1", type: "health", checked: [] }
        ]}
      />
    );
    dashboard.find(".btn-primary").simulate("click");
    dashboard.find(".btn-secondary").simulate("click");
    expect(dashboard.find(".habit-row").length).toBe(1);
  });
});
