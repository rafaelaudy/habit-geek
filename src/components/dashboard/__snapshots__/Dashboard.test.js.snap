// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard component Renders Dashboard list 1`] = `
<DashboardView
  currentHabits={Array []}
  currentWeek="y1w1"
  onAddNewHabit={[Function]}
  onUpdateHabit={[Function]}
  previousHabits={Array []}
  previousWeek="y1w0"
  toggleDayHabit={[Function]}
  username="rafa"
/>
`;

exports[`Dashboard component Renders New Habit 1`] = `
<Dashboard
  currentHabits={Array []}
  currentWeek="y1w1"
  deleteHabit={[Function]}
  previousHabits={Array []}
  previousWeek="y1w0"
  saveHabit={[Function]}
  toggleDayHabit={[Function]}
  username="rafa"
>
  <SaveHabit
    deleteHabit={[Function]}
    onGoBack={[Function]}
    saveHabit={[Function]}
  >
    <form
      className="mobile-size-container "
      noValidate={true}
      onSubmit={[Function]}
    >
      <h2>
        save-habit-new-title
      </h2>
      <div
        className="mb-3"
      >
        <label
          htmlFor="new-habit-name"
        >
          save-habit-name
        </label>
        <input
          className="form-control"
          id="new-habit-name"
          onChange={[Function]}
          required={true}
          value=""
        />
        <div
          className="invalid-feedback"
        >
          save-habit-name-validation
        </div>
      </div>
      <div
        className="mb-3"
      >
        <label
          htmlFor="new-habit-type"
        >
          save-habit-type
        </label>
        <select
          className="form-control"
          id="new-habit-type"
          onBlur={[Function]}
          onChange={[Function]}
          required={true}
          value=""
        >
          <option
            value=""
          />
          <option
            key="save-habit-type-option-Health"
            value="Health"
          >
            Health
          </option>
          <option
            key="save-habit-type-option-Social"
            value="Social"
          >
            Social
          </option>
          <option
            key="save-habit-type-option-Career"
            value="Career"
          >
            Career
          </option>
          <option
            key="save-habit-type-option-Hobbies"
            value="Hobbies"
          >
            Hobbies
          </option>
        </select>
        <div
          className="invalid-feedback"
        >
          save-habit-type-validation
        </div>
      </div>
      <div
        className="mb-3"
      >
        <label
          htmlFor="new-habit-frequency"
        >
          save-habit-frequency
        </label>
        <select
          className="form-control"
          id="new-habit-frequency"
          onBlur={[Function]}
          onChange={[Function]}
          required={true}
          value=""
        >
          <option
            value=""
          />
          <option
            value="1"
          >
            1x
          </option>
          <option
            value="2"
          >
            2x
          </option>
          <option
            value="3"
          >
            3x
          </option>
          <option
            value="4"
          >
            4x
          </option>
          <option
            value="5"
          >
            5x
          </option>
          <option
            value="6"
          >
            6x
          </option>
          <option
            value="7"
          >
            7x
          </option>
        </select>
        <div
          className="invalid-feedback"
        >
          save-habit-frequency-validation
        </div>
      </div>
      <hr
        className="mb-4"
      />
      <button
        className="btn btn-primary btn-lg btn-block"
      >
        save-habit-save
      </button>
      <button
        className="btn btn-secondary btn-lg btn-block"
        onClick={[Function]}
        type="button"
      >
        save-habit-cancel
      </button>
    </form>
  </SaveHabit>
</Dashboard>
`;
